### PR TITLE
feat: use v0.1.1 of alloy from crates.io

### LINKED ISSUE #125: feat: use a published version of alloy
We are currently using `alloy` via a git dependency, which is not ideal. For example, we wouldn't be able to publish crates depending on it.

Now that there's a published version of alloy on crates.io, we should start using it.

### PR DESCRIPTION
**Pull request by alexfertel**
Resolves #125

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -38,24 +38,26 @@ keywords = ["arbitrum", "ethereum", "stylus"]
...
   "provider-http",
   "rpc-client",
   "rpc-types-eth",
-  "signers",
-  "signer-wallet",
+  "signer-local",
   "getrandom",
 ] }
+# Even though `alloy` includes `alloy-primitives` and `alloy-sol-types` we need
+# to keep both versions for compatibility with the SDK. Once they start using
```
Comment (ggonzalez94) [inline Cargo.toml:57]:
```suggestion
# to keep both versions for compatibility with the Stylus SDK. Once they start using
```

**Review by ggonzalez94 — APPROVED** [review]
LGTM! Less than a day since the version was published and we already have a PR 🚀

**Review by bidzyyys — APPROVED** [review]
LGTM 🟢 🚀

**Review by qalisander — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (alexfertel) [reply Cargo.toml:57]: ooops, missed this one. I'll add it in a separate PR.

**Pull request merged**